src/Microsoft.DotNet.Build.Tasks/PackageFiles/VersionTools.targets

```diff
@@ -66,22 +63,47 @@
+    Update to the latest dependencies available remotely, then submit a pull request.
+  -->
+  <Target Name="UpdateToRemoteDependenciesAndSubmitPullRequest">
+    <UpdateToRemoteDependenciesAndSubmitPullRequest DependencyBuildInfo="@(DependencyBuildInfo)"
+                                                    ProjectJsonFiles="@(ProjectJsonFiles)"
+                                                    UpdateStep="@(UpdateStep)"
+                                                    ProjectRepoName="$(ProjectRepoName)"
+                                                    ProjectRepoOwner="$(ProjectRepoOwner)"
+                                                    ProjectRepoBranch="$(ProjectRepoBranch)"
+                                                    GitHubAuthToken="$(GitHubAuthToken)"
+                                                    GitHubUser="$(GitHubUser)"
+                                                    GitHubEmail="$(GitHubEmail)"
+                                                    GitHubAuthor="$(GitHubAuthor)"
+                                                    NotifyGitHubUsers="@(NotifyGitHubUsers)"
+                                                    CurrentRefXmlPath="$(CurrentRefXmlPath)"
+                                                    AlwaysCreateNewPullRequest="$(AlwaysCreateNewPullRequest)"
+                                                    CommitMessage="$(CommitMessage)" />
+  </Target>
 
-  <Target Name="UpdateDependenciesAndSubmitPullRequest">
-    <UpdateDependenciesAndSubmitPullRequest DependencyBuildInfo="@(DependencyBuildInfo)"
-                                            ProjectJsonFiles="@(ProjectJsonFiles)"
-                                            UpdateStep="@(UpdateStep)"
-                                            ProjectRepoName="$(ProjectRepoName)"
-                                            ProjectRepoOwner="$(ProjectRepoOwner)"
-                                            ProjectRepoBranch="$(ProjectRepoBranch)"
-                                            GitHubAuthToken="$(GitHubAuthToken)"
-                                            GitHubUser="$(GitHubUser)"
-                                            GitHubEmail="$(GitHubEmail)"
-                                            GitHubAuthor="$(GitHubAuthor)"
-                                            NotifyGitHubUsers="@(NotifyGitHubUsers)"
-                                            CurrentRefXmlPath="$(CurrentRefXmlPath)"
-                                            AlwaysCreateNewPullRequest="$(AlwaysCreateNewPullRequest)" />
+  <!--
+    Deprecated auto-upgrade PR target. Redirects to the target's new name.
+  -->
+  <Target Name="UpdateDependenciesAndSubmitPullRequest"
+          DependsOnTargets="UpdateToRemoteDependenciesAndSubmitPullRequest">
+    <Message Text="'UpdateDependenciesAndSubmitPullRequest' has been renamed to 'UpdateToRemoteDependenciesAndSubmitPullRequest'. The old name is deprecated. Ran the new target instead."
```

Comment:
I don't think it is worth breaking usages of this. Instead I would just keep it working or just stick with this name.

Reply:
Nevermind it does continue to work you just have a message. That's reasonable.

Reply:
Why the change at all?  Just to add the word `remote`?  Is it necessary?

Reply:
Here are the steps that landed me here:

 1. There needs to be a way for a dev to run an update that actually pulls in the remote status--this is specifically a goal for submodule updating. `UpdateDependenciesAndSubmitPullRequest` had that logic embedded inside, so I refactored it into a `UpdateToRemoteDependencies` target. Then I had:
    * `UpdateDependencies`
    * `UpdateToRemoteDependencies`
    * `UpdateDependenciesAndSubmitPullRequest`
 2. Now `UpdateDependenciesAndSubmitPullRequest` is misleading. It's actually adding onto `UpdateToRemoteDependencies`, so I renamed it with a back-compat target:
    * `UpdateDependencies`
    * `UpdateToRemoteDependencies`
    * `UpdateToRemoteDependenciesAndSubmitPullRequest`
    * (`deprecated`) `UpdateDependenciesAndSubmitPullRequest`

The `...And` is the source of the clunkiness here. I've kept it so far because it seems more reasonable to keep it than try to export all the info we need from `UpdateToRemoteDependencies` and stuff it back into a "`SubmitPullRequest`" task. I haven't seriously considered making that split, though... now that I'm thinking about it, it actually seems like it could be reasonable. I'll take a stab at it when I'm done with the other comments.